hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/regionreplication/RegionReplicationSink.java

```diff
@@ -448,4 +458,35 @@ public void waitUntilStopped() throws InterruptedException {
       }
     }
   }
+
+  @RestrictedApi(explanation = "Should only be called in tests", link = "",
+      allowedOnPath = ".*/src/test/.*")
+  RegionReplicationFlushRequester getFlushRequester() {
+    return this.flushRequester;
+  }
+
+  @RestrictedApi(explanation = "Should only be called in tests", link = "",
+      allowedOnPath = ".*/src/test/.*")
+  IntHashSet getFailedReplicas() {
+    synchronized (entries) {
+      return this.failedReplicas;
+    }
+  }
+
+  @RestrictedApi(explanation = "Should only be called in tests", link = "",
+      allowedOnPath = ".*/src/test/.*")
+  boolean isSending() {
+    synchronized (entries) {
+      return this.sending;
+    }
+  }
+
+  @RestrictedApi(explanation = "Should only be called in tests", link = "",
+      allowedOnPath = ".*/src/test/.*")
+  Queue<SinkEntry> getEntries() {
+    synchronized (entries) {
```

Comment:
We just return the entries out so I do not get the point on 'synchronized (entries)' here...